core/src/mock_alpenglow_consensus.rs

```diff
@@ -0,0 +1,741 @@
+                VotorMessageType::try_from(i as u64).unwrap(), // this unwrap is ok since i is in static range
+                percent_collected[i],
+                stake_weighted_delay[i]
+            );
+        }
+        datapoint_info!(
+            "mock_alpenglow",
+            ("total_peers", peers.len(), f64),
+            ("packets_collected_notarize", total_voted_nodes[0], f64),
+            (
+                "percent_stake_collected_notarize",
+                percent_collected[0],
+                f64
+            ),
+            ("weighted_delay_ms_notarize", stake_weighted_delay[0], f64),
+            ("packets_collected_notarize_cert", total_voted_nodes[1], f64),
+            (
+                "percent_stake_collected_notarize_cert",
+                percent_collected[1],
+                f64
+            ),
+            (
+                "weighted_delay_ms_notarize_cert",
+                stake_weighted_delay[1],
+                f64
+            ),
+            ("packets_collected_finalize", total_voted_nodes[2], f64),
+            (
+                "percent_stake_collected_finalize",
+                percent_collected[2],
+                f64
+            ),
+            ("weighted_delay_ms_finalize", stake_weighted_delay[2], f64),
+            ("packets_collected_finalize_cert", total_voted_nodes[3], f64),
+            (
+                "percent_stake_collected_finalize_cert",
+                percent_collected[3],
+                f64
+            ),
+            (
+                "weighted_delay_ms_finalize_cert",
+                stake_weighted_delay[3],
+                f64
+            ),
+        );
+    }
+
+    pub(crate) fn join(self) -> thread::Result<()> {
+        self.should_exit.store(true, Ordering::Relaxed);
+        self.sender_thread.join()?;
+        self.listener_thread.join()
+    }
+}
+
+fn prep_and_sign_packet(
+    packet_buf: &mut [u8],
+    slot: Slot,
+    id: &Pubkey,
+    state: VotorMessageType,
+    keypair: &Keypair,
+) {
+    // prepare the packet to send and sign it
+    {
+        let pkt = MockVotePacketHeader::from_bytes_mut(packet_buf);
+        pkt.slot_number = slot;
+        pkt.sender = *id.as_array();
+        pkt.signature = [0; SIGNATURE_BYTES];
+        pkt.state = state as u64;
+    }
+    let signature = keypair.sign_message(&packet_buf[SIGNATURE_BYTES..MOCK_VOTE_HEADER_SIZE]);
+    {
+        let pkt = MockVotePacketHeader::from_bytes_mut(packet_buf);
+        pkt.signature = *signature.as_array();
+    }
+}
+
+mod control_pubkey {
+    solana_pubkey::declare_id!("9PsiyXopc2M9DMEmsEeafNHHHAUmPKe9mHYgrk6fHPyx");
+}
+
+#[derive(Deserialize, Debug)]
+#[repr(C)]
+pub(crate) struct TestConfig {
+    _version: u8,         // This is part of Record program header
+    _authority: [u8; 32], // This is part of Record program header
+    test_interval_slots: u8,
+    verify_signatures: bool,
+    packet_size: u16, // packet size
+    _future_use: [u8; 16],
```

Comment:
might be good to also configure:

1. how many packets are sent per slot
2. interval in ms at which these packets are sent 

@AshwinSekar  we can have up to 3 notar/notar-fallbacks and a finalization in short order right?

Reply:
We can, but it should be extremely rare. I think Alex is mostly testing the normal case here where we send Notarize and send Finalize and send certs as well.

We are starting this test because people claimed that even under happy case we will kill some routers in the Internet. So I think we can just start with the normal case testing at first. This test code is already pretty complex, we can maybe leave 3 notar/notar-fallbacks to next version?